platform/jewel/ui/src/main/kotlin/org/jetbrains/jewel/ui/component/SimpleListItem.kt

```diff
@@ -12,23 +12,218 @@ import androidx.compose.runtime.Composable
 import androidx.compose.runtime.remember
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.ColorFilter
 import androidx.compose.ui.semantics.selected
 import androidx.compose.ui.semantics.semantics
 import androidx.compose.ui.text.style.TextOverflow
 import androidx.compose.ui.unit.Dp
 import androidx.compose.ui.unit.dp
+import org.jetbrains.annotations.ApiStatus.ScheduledForRemoval
 import org.jetbrains.jewel.foundation.GenerateDataFunctions
 import org.jetbrains.jewel.foundation.theme.JewelTheme
 import org.jetbrains.jewel.ui.component.styling.SimpleListItemStyle
 import org.jetbrains.jewel.ui.icon.IconKey
+import org.jetbrains.jewel.ui.painter.PainterHint
 import org.jetbrains.jewel.ui.theme.simpleListItemStyle
 
+/**
+ * A simple list item layout comprising of a content slot and an optional icon to its start side.
+ *
+ * The text will only take up one line and is ellipsized if too long to fit.
+ *
+ * @param text The text displayed in the list item
+ * @param state The state of the list item, containing selection and activity status.
+ * @param colorFilter Optional [ColorFilter] applied to the item, typically used with the icon.
+ * @param modifier Optional [Modifier] to be applied to the list item.
+ * @param textModifier The [Modifier] applied to the list item text
+ * @param iconModifier Optional [Modifier] to be applied specifically to the icon.
+ * @param icon Optional [IconKey] representing the icon displayed in the list item.
+ * @param iconContentDescription Content description for accessibility purposes for the icon.
+ * @param style Optional [SimpleListItemStyle] for defining the appearance of the list item.
+ * @param height The height of the list item; default is based on the theme's global metrics.
+ * @param painterHints Optional vararg of [PainterHint] to provide hints for painting customizations.
+ */
+@Composable
+public fun SimpleListItem(
+    text: String,
+    state: ListItemState,
+    colorFilter: ColorFilter?,
```

Comment:
This should have a default `null` value (same in other overloads); put it right before `painterHints`, like the `Icon` API does that I pointed to as reference